### PR TITLE
Fix NativeAOT unhandled exception stack trace

### LINKED ISSUE #91298: [native aot] Unhandled exceptions are not printed to console
Compile and run with nativeaot on current main:

```
throw new Exception("My unhandled exception");
```

## Expected result

Unhandled exception message is printed to console.

## Actual result

No message printed to console

### PR DESCRIPTION
**Pull request by janvorli**
The recent change to add a new exception handling mechanism to coreclr has broken stack traces on unhandled exceptions in nativeaot.

This change fixes it by reverting the part of the change that caused the problem and that turned out to not to be needed for the new exception handling either.

Close #91298

### CODE REVIEW

**Comment by agocke** [comment]
Should we add a smoke test for this?

**Review by jkotas — APPROVED** [review]
Thanks!

**Comment by jkotas** [comment]
> Should we add a smoke test for this?

I think a regular test - that runs on all runtime flavors - would be enough.

**Comment by janvorli** [comment]
Ok, let me add a test here.

File: src/tests/baseservices/exceptions/unhandled/unhandled.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                throw new Exception("Test");
+            }
+
+            List<string> lines = new List<string>();
+
+            Process testProcess = new Process();
+
+            testProcess.StartInfo.FileName = Environment.ProcessPath;
+            testProcess.StartInfo.Arguments = Environment.CommandLine + " throw";
+            testProcess.StartInfo.UseShellExecute = false;
```
Comment (jkotas) [inline src/tests/baseservices/exceptions/unhandled/unhandled.cs:27]:
```suggestion
```
Unnecessary - it is the default in .NET Core.

File: src/tests/baseservices/exceptions/unhandled/unhandled.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                }
+            };
+
+            testProcess.Start();
+            testProcess.BeginErrorReadLine();
+            testProcess.WaitForExit();
+            testProcess.CancelErrorRead();
+
+            int expectedExitCode;
+            if ((Environment.OSVersion.Platform == PlatformID.Unix) || (Environment.OSVersion.Platform == PlatformID.MacOSX))
```
Comment (jkotas) [inline src/tests/baseservices/exceptions/unhandled/unhandled.cs:44]:
```suggestion
            if (!OperatingSystem.IsWindows())
```
`Environment.OSVersion.Platform` is a soft-obsolete API. It behaves in non-sensical way for historic reasons (for example, it returnns `PlatformID.Unix` even on macOS).

File: src/tests/baseservices/exceptions/unhandled/unhandled.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                expectedExitCode = unchecked((int)0xE0434352);
+            }
+
+            if (expectedExitCode != testProcess.ExitCode)
+            {
+                Console.WriteLine($"Wrong exit code 0x{testProcess.ExitCode:X8}");
+                return 101;
+            }
+
+            if (Regex.Match(lines[0], @"Unhandled exception[.:] System\.Exception\: Test", RegexOptions.IgnoreCase) == Match.Empty)
```
Comment (jkotas) [inline src/tests/baseservices/exceptions/unhandled/unhandled.cs:63]: Feel free to change the native AOT string to `Unhandled exception` and use a simple exact match here instead.

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**